ui.frontend/src/utils.js

```diff
@@ -177,6 +184,32 @@ class Utils {
             Utils.registerMutationObserver(formContainer, fieldCreator, fieldSelector, fieldClass);
         });
     }
+
+    /**
+     * Register time based auto save
+     * @param {module:FormView~FormContainer} formContainer - The form container.
+     * @param autoSaveProperties
+     */
+    static setupAutoSave(formContainer, autoSaveProperties) {
+        const enableAutoSave = autoSaveProperties['fd:enableAutoSave'];
+        if (enableAutoSave) {
+            const autoSaveStrategyType = autoSaveProperties['fd:autoSaveStrategyType'];
+            const autoSaveInterval = autoSaveProperties['fd:autoSaveInterval'];
+            const formModel = formContainer.getModel();
+            const saveEndPoint = Utils.getContextPath() + '/adobe/forms/af/save/' + formModel.id;
+            if (autoSaveStrategyType === 'time' && autoSaveInterval) {
+                if (Utils.#autoSaveIntervalId === -1) {
+                    console.log("Registering time based auto save");
+                    Utils.#autoSaveIntervalId = setInterval(() => {
+                        formModel.dispatch(new Save({
+                            'action': saveEndPoint
+                        }));
```

Comment:
Isn't saveAction  already defined in the definition? Why do we need pass it again in the event?

Reply:
No, we have not defined saveAction in form definition. At time of rule, we are passing in rule only.